src/Parser.php

```diff
@@ -679,6 +655,69 @@ private function parseTableRows(): array
         return $table;
     }
 
+    /**
+     * @param TTitleToken $keywordToken
+     *
+     * @return array{title:string|null, description:string|null}
+     */
+    private function parseTitleAndDescription(array $keywordToken): array
+    {
+        $originalTitle = trim($keywordToken['value'] ?? '');
+        $textLines = [];
+
+        while (in_array($predicted = $this->predictTokenType(), ['Newline', 'Text', 'Comment'])) {
+            $token = $this->expectTokenType($predicted);
+
+            if ($token['type'] === 'Comment') {
+                continue;
+            }
+
+            $text = match ($token['type']) {
+                'Newline' => '',
+                'Text' => $token['value'],
+                default => throw new LogicException('Unexpected token type: ' . $token['type']),
+            };
+
+            // The only time we use $token['value'] is if we got a `Text` token.
+            // ->expectTokenType('Text') is tagged as returning a `TStringValueToken`, where 'value' cannot be null
+            // However PHPStan cannot follow the chain through predictTokenType -> expectTokenType -> $token['type']
+            assert($text !== null, 'Text token value should not be null');
```

Comment:
Without this, PHPStan was complaining that the `$text` argument to preg_replace could be null (when in fact it cannot be). I'm not sure if there's a cleverer way to solve this.